uaclient/cli/fix.py

```diff
@@ -52,6 +104,747 @@ def fix_parser(parser):
+                return True
+
+            if _prompt_for_enable(cfg, ent.name):
+                return True
+            else:
+                print(
+                    messages.SECURITY_UA_SERVICE_NOT_ENABLED.format(
+                        service=ent.name
+                    )
+                )
+
+        else:
+            print(
+                messages.SECURITY_UA_SERVICE_NOT_ENTITLED.format(
+                    service=ent.name
+                )
+            )
+
+    return False
+
+
+def _handle_fix_status_message(
+    status: FixStatus, issue_id: str, extra_info: str = ""
+):
+    if status == FixStatus.SYSTEM_NON_VULNERABLE:
+        print(
+            util.handle_unicode_characters(
+                messages.SECURITY_ISSUE_RESOLVED.format(
+                    issue=issue_id, extra_info=extra_info
+                )
+            )
+        )
+    elif status == FixStatus.SYSTEM_NOT_AFFECTED:
+        print(
+            util.handle_unicode_characters(
+                messages.SECURITY_ISSUE_UNAFFECTED.format(
+                    issue=issue_id, extra_info=extra_info
+                )
+            )
+        )
+    elif status == FixStatus.SYSTEM_VULNERABLE_UNTIL_REBOOT:
+        print(
+            util.handle_unicode_characters(
+                messages.SECURITY_ISSUE_NOT_RESOLVED.format(
+                    issue=issue_id, extra_info=extra_info
+                )
+            )
+        )
+    else:
+        print(
+            util.handle_unicode_characters(
+                messages.SECURITY_ISSUE_NOT_RESOLVED.format(
+                    issue=issue_id, extra_info=extra_info
+                )
+            )
+        )
+
+
+def execute_fix_plan(
+    fix_plan: FixPlanResult, dry_run: bool, cfg: UAConfig
+) -> Tuple[FixStatus, List[security.UnfixedPackage]]:
+    full_plan = [
+        *fix_plan.plan,
+        *fix_plan.warnings,
+    ]  # type: List[Union[FixPlanStep, FixPlanWarning]]
+
+    pkg_index = 0
+    unfixed_pkgs = []  # type: List[security.UnfixedPackage]
+    installed_pkgs = set()  # type: Set[str]
+    fix_status = FixStatus.SYSTEM_NON_VULNERABLE
+    affected_pkgs = fix_plan.affected_packages or []
+    print_pkg_header = True
+    warn_package_cannot_be_installed = False
+
+    if affected_pkgs:
+        if len(affected_pkgs) == 1:
+            plural_str = " is"
+        else:
+            plural_str = "s are"
+
+        msg = (
+            messages.SECURITY_AFFECTED_PKGS.format(
+                count=len(affected_pkgs), plural_str=plural_str
+            )
+            + ": "
+            + ", ".join(sorted(affected_pkgs))
+        )
+        print(
+            textwrap.fill(
+                msg,
+                width=PRINT_WRAP_WIDTH,
+                subsequent_indent="    ",
+                replace_whitespace=False,
+            )
+        )
+
+    for step in sorted(full_plan, key=lambda x: x.order):
+        if isinstance(step, FixPlanWarningPackageCannotBeInstalled):
```

Comment:
An idea: Do you think it would be any easier to read if there was a function for each step type? So this loop would look more like this:
```python
fix_context = FixContext(all the shared state that need to be updated by each step)
for step in steps:
    if isinstance(step, Klass):
        execute_klass_step(fix_context, step)
    elif isinstance(step, Klass2):
        execute_klass2_step(fix_context, step)
    ....
```